resolver/resolver.cc

```diff
@@ -948,10 +948,75 @@ class ResolveConstantsWalk {
             transformAncestor(ctx.withOwner(klass), singleton, ancst, isInclude);
         }
 
+        // TODO: Only do this check in --stripe-mode. Currently it does the check but skips reporting the error if
+        // not in --stripe-mode.
+        //
+        // TODO: Do rigorous perf analysis.
+        if (!ctx.file.data(ctx).isPackage()) {
+            const auto ambigDef = findAnyDefinitionAmbiguousWithCurrent(ctx);
+            if (ambigDef.has_value()) {
+                if (auto e = ctx.beginError(original.loc, core::errors::Resolver::AmbiguousDefinitionError)) {
+                    e.setHeader("Class definition is ambiguous", klass.show(ctx));
+                    e.addErrorLine((*ambigDef).loc(ctx), "Previously defined here.");
+                }
+            }
+        }
+
         nesting_ = nesting_->parent;
         return tree;
     }
 
+    const std::optional<core::SymbolRef> findAnyDefinitionAmbiguousWithCurrent(core::Context ctx) {
```

Comment:
Is there a reason we can't return just `core::SymbolRef` here, with a default-constructed (i.e. `!exists()`) `SymbolRef` to indicate non-ambiguity?